commonmark-extensions/test/pipe_tables.md

```diff
@@ -312,3 +312,126 @@ Tables can be nested in other elements, but don't benefit from laziness.
 ----|-----</p>
 </blockquote>
 ````````````````````````````````
+
+
+As a special case, pipes in inline code in tables are escaped
+with backslashes.
+
+The parsing rule for CommonMark is that block structures are parsed before
+inline structures are. Normally, this means backslashes aren't allowed to have
+any effect on block structures at all. Tables do consider backslashes, but
+not the same way inline syntax does: they have higher precedence than anything
+else, as-if they were parsed in a completely separate pass.
```

Comment:
I would just suggest making it a bit clearer *exactly* what the rule is.  This text might make one think that first, ALL backslash escapes are resolved (even in code backticks), and then the line is divided into cells which are parsed normally. But in fact, it's only backslashes that occur right before a `|` character that are resolved in this first pass -- correct?

Reply:
That sounds good, but maybe we need to make it clear that there are really two passes: an initial pass in which nothing happens but that `\|` is replaced with `|`, even if it occurs in backticks, then, after the table cells have been segmented (splitting on unescaped `|`), regular inline parsing that treats `\` as a backslash escape where it usually would.